kernel/mmu.c

```diff
@@ -722,11 +722,8 @@ void k_mem_unmap_phys_guard(void *addr, size_t size, bool is_anon)
 #else
 			ret = arch_page_phys_get(pos, &phys);
 #endif
-			__ASSERT(ret == 0,
-				 "%s: cannot unmap an unmapped address %p",
-				 __func__, pos);
```

Comment:
This is here to catch coding error. If a continuously mapped virtual memory space has an unmapped hole, this means someone who should not be poking at this region has unmapped some parts of it. The original owner/mapper of this region has no idea that there is a hole and will continue accessing it, resulting in unexpected page faults.

What exactly are you trying to solve here?

Reply:
Originally I used some other call to check if a region was mapped. If the call failed, I knew not to call `k_mem_unmap()`. @npitre said not use the call to check if the region was mapped and to simply call `k_mem_unmap()` and that this function would simply return if the region was unmapped.

That is not the observed behaviour. The observed behaviour is that the assert is always triggered now.

There seems to be an impedance mismatch between how maintainers expect this function to work.

That impedance mismatch is unfortunately blocking quite a lot features for 3.7.0 have otherwise been ready to merge for quite a while now.

I really hate asking for an extension after feature freeze, but likely have no choice at this point.